silkrpc/core/evm_executor.cpp

```diff
@@ -168,34 +168,51 @@ asio::awaitable<ExecutionResult> EVMExecutor::call(const silkworm::Block& block,
 
                 const intx::uint256 base_fee_per_gas{evm.block().header.base_fee_per_gas.value_or(0)};
                 const intx::uint256 effective_gas_price{txn.effective_gas_price(base_fee_per_gas)};
-                state.subtract_from_balance(*txn.from, txn.gas_limit * effective_gas_price);
-
-                if (txn.to.has_value()) {
-                    state.access_account(*txn.to);
-                    // EVM itself increments the nonce for contract creation
-                    state.set_nonce(*txn.from, txn.nonce + 1);
-                }
-
-                for (const silkworm::AccessListEntry& ae : txn.access_list) {
-                    state.access_account(ae.account);
-                    for (const evmc::bytes32& key : ae.storage_keys) {
-                        state.access_storage(ae.account, key);
-                    }
+                const auto have = state.get_balance(*txn.from);
+                const auto want = txn.gas_limit * effective_gas_price;
+
+                if (have < want) {
+                   silkworm::Bytes data{};
+                   std::string s1 = "insufficient funds for gas * price + value: address 0x";
+                   std::string addrValue = silkworm::to_hex(*txn.from);
+                   std::string haveStr = " have ";
+                   std::string haveValue = intx::to_string(have);
+                   std::string wantStr = " want ";
+                   std::string wantValue = intx::to_string(want);
+                   std::string error_message = s1 + addrValue + haveStr + haveValue + wantStr + wantValue;
```

Comment:
I would replace lines 176-182 with:
```
std::string from = silkworm::to_hex(*txn.from);
std::string error = "insufficient funds for gas * price + value: address 0x" + from + " have " + intx::to_string(have) + " want " + intx::to_string(want);
```
to improve overall readability of method: the line becomes quite long but I would keep the error message formatting as compact as possible because it is not so relevant

Reply:
done